packages/tests/tests/store/message_hash.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { DecodedMessage } from "@waku/core";
+import type { LightNode } from "@waku/interfaces";
+import { messageHash } from "@waku/message-hash";
+import { assert } from "chai";
+
+import {
+  afterEachCustom,
+  beforeEachCustom,
+  ServiceNode,
+  tearDownNodes
+} from "../../src/index.js";
+
+import {
+  runStoreNodes,
+  sendMessages,
+  TestDecoder,
+  TestShardInfo,
+  totalMsgs
+} from "./utils.js";
+
+describe("Waku Store, message hash query", function () {
+  this.timeout(15000);
+  let waku: LightNode;
+  let nwaku: ServiceNode;
+
+  beforeEachCustom(this, async () => {
+    [nwaku, waku] = await runStoreNodes(this.ctx, TestShardInfo);
+  });
+
+  afterEachCustom(this, async () => {
+    await tearDownNodes(nwaku, [waku]);
+  });
+
+  it("can query messages by message hash", async function () {
+    // Send messages first
+    await sendMessages(
+      nwaku,
+      totalMsgs,
+      TestDecoder.contentTopic,
+      TestDecoder.pubsubTopic
+    );
+
+    // Generate message hashes for the test
+    const messageHashes: Uint8Array[] = [];
+
+    // Create message hashes for all numbers from 0 to totalMsgs-1, matching the payload pattern in sendMessages
+    for (let i = 0; i < totalMsgs; i++) {
+      // Using type assertion to handle type mismatch
+      messageHashes.push(
+        messageHash(TestDecoder.pubsubTopic, {
+          payload: new Uint8Array([i]) as any,
+          contentTopic: TestDecoder.contentTopic,
+          version: undefined,
+          timestamp: undefined,
+          meta: undefined,
+          rateLimitProof: undefined,
+          ephemeral: undefined
+        })
+      );
+    }
+
+    // Query messages by hash only - DO NOT use contentTopics or other filters here
+    const messages: DecodedMessage[] = [];
+    // When using messageHashes, do NOT include ANY content filter properties
+    for await (const page of waku.store.queryGenerator([TestDecoder], {
+      messageHashes: messageHashes,
+      pubsubTopic: TestDecoder.pubsubTopic
+    })) {
+      for await (const msg of page) {
+        messages.push(msg as DecodedMessage);
+      }
+    }
+
+    // Note: The real issue might be that message hash lookup is not properly supported
```

Comment:
I am missing the value of the test here. It would succeed even if no messages are found, right?